src/mongoose_riak.erl

```diff
@@ -55,26 +55,20 @@ start() ->
         undefined ->
             ignore;
         RiakOpts ->
+            ?WARNING_MSG("Deprecated riak_server option, please use outgoing_pools", []),
             start_pool(RiakOpts)
     end.
 
 start_pool(RiakOpts) ->
-    {_, RiakAddr} = get_riak_opt(address, RiakOpts),
-    {_, RiakPort} = get_riak_opt(port, RiakOpts),
-    {_, Workers} = get_riak_opt(pool_size, RiakOpts, {pool_size, 20}),
-    SecurityOptsKeys = [credentials, cacertfile, ssl_opts],
-    SecurityOpts = [get_riak_opt(OptKey, RiakOpts) || OptKey <- SecurityOptsKeys],
-    RiakPBOpts = [auto_reconnect, keepalive],
-    WorkerArgs = maybe_add_additional_opts(RiakPBOpts, SecurityOpts),
-    Worker = {riakc_pb_socket, [RiakAddr, RiakPort, WorkerArgs]},
+    {_, Workers} = mongoose_wpool_riak:get_riak_opt(pool_size, RiakOpts, {pool_size, 20}),
     %% TODO (Bartek Gorny): improve worker_pool, then use available_worker strategy here
```

Comment:
@bartekgorny I feel this comment has not aged well. What does it mean? Why would we want to use `available_worker`?